django/core/management/commands/shell.py

```diff
@@ -111,10 +115,44 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_and_report_namespace(self, verbosity):
+        namespace = self.get_namespace()
+
+        if verbosity == 1:
+            self.stdout.write(
+                f"{len(namespace)} objects imported automatically",
+                self.style.SUCCESS,
+            )
+        elif verbosity >= 2:
+            for obj_name, obj in namespace.items():
+                if hasattr(obj, "__module__") and hasattr(obj, "__qualname__"):
+                    if getattr(obj, "__qualname__").find(".") == -1:
+                        self.stdout.write(
+                            f"from {obj.__module__} import {obj_name}",
+                            self.style.SUCCESS,
+                        )
+
```

Comment:
```suggestion
            imports_by_module = {}
            for obj_name, obj in namespace.items():
                if (
                    hasattr(obj, "__module__") and
                    hasattr(obj, "__qualname__") and
                    obj.__qualname__.find(".") == -1
                ):
                    module = obj.__module__
                    collected_imports = imports_by_module.get(module, [])
                    imports_by_module[module] = collected_imports + [obj_name]
            for module, imported_objects in imports_by_module.items():
                self.stdout.write(
                    f"from {module} import {', '.join(imported_objects)}",
                    self.style.SUCCESS,
                )
```
We might be able to reduce the number of lines printed by having one line per module rather than per object